Assets/Project/Scripts/ObstacleManager.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Collections;
+using UnityEngine;
+
+namespace Assets.Project.Scripts.Obstacles
+{
+    public class ObstacleManager : MonoBehaviour
+    {
+        #region Fields
+        [SerializeField] private GameObject[] allObstacles; // list of prefabs outside
+        [SerializeField] private float fastestSpawnRate;
+        [SerializeField] private float lowestSpawnRate;
+        #endregion
+
+        private void Start()
+        {
+            // init random spawning coroutine
+            fastestSpawnRate = 0.2f;
+            lowestSpawnRate = 1f;
+            StartCoroutine(RandomSpawning(1));            
+        }
+
+        private void Update()
+        {
+
+        }
+
+        // random spawn coroutine function that changes the spawn rate every time
+        private IEnumerator RandomSpawning(float spawnInterval)
+        {         
+            yield return new WaitForSeconds(spawnInterval);
+            while (true)
+            {
+                SpawnObstacle();
+
+                Debug.Log($"Spawning every {spawnInterval}s");
+                spawnInterval = Random.Range(fastestSpawnRate, lowestSpawnRate);
+                yield return new WaitForSeconds(spawnInterval);
+            }            
+        }
+        
+        // spawn obstacle
+        private void SpawnObstacle()
+        {
+            int refFreeObstacle = -1;
+            if (TryGetAvailableObstacle(out refFreeObstacle))
+            {
+                Debug.Log("Spawning");
```

Comment:
Either delete the Debug.Logs or put it in an if UNITY_EDITOR region in case you forget to remove it later on